### ISSUE TITLE
Error on clean install

### ISSUE THREAD

**Issue by thany** [vercel/next.js#33234]
### Run `next info` (available from version 12.0.8 and up)

    Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Pro
    Binaries:
      Node: 16.13.1
      npm: 8.1.2
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 12.0.8
      react: 17.0.2
      react-dom: 17.0.2

### What version of Next.js are you using?

12.0.8

### What version of Node.js are you using?

16.13.1

### What browser are you using?

Firefox 95

### What operating system are you using?

Windows 10 21H2

### How are you deploying your application?

I'm not

### Describe the Bug

Error appears after a **clean install** with typescript:

```
Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
[etc]
```

### Expected Behavior

No error

### To Reproduce

Run the following command and give it any project name you haven't created before:

```
npx create-next-app@latest --typescript
```

Then, start it:

```
npm run dev
```

This will produce the error.

You might need to install on a different drive than whatever you did above, in order to trigger the error. This seems to be the crux.

**Comment by thany** [comment]
See also [this discussion](https://github.com/vercel/next.js/discussions/32034).

Turns out this is really, genuinely, honestly, absolutely, a bug. Not a misconfig. Not a problem in our pc's. Not a problem between the keyboard & chair. Not a problem anywhere else but in Next.js or one of its dependencies.

Please note, you as a Next.js author might not be able to reproduce this. Though luck. I can reproduce it 100% of the time, and I'd be happy to share any detailed information on this when I can. But do ASK for it - I don't magically know what you might need.

**Comment by thany** [comment]
Here's the browser console (this repeats):

![image](https://user-images.githubusercontent.com/152227/149227236-a9aa8d2f-56a6-4d64-9241-081f51c5b542.png)

And a tiny fraction of the CLI getting absolutely flooded with errors:

![image](https://user-images.githubusercontent.com/152227/149227896-a78f4631-3107-4f73-8378-96d33b922c1d.png)

**Comment by balazsorban44** [comment]
Hi, I could not reproduce this with your reproduction. 

Sounds like a rogue `react` package is interfering with your project.

This kind of issue can happen if you have used `yarn link` or `npm link`.

Does this reproduce on multiple machines, or only on yours? Since you mention "our pc's", could you please check this on other devices? That would indeed be helpful since we could narrow this down to either OS or user config level.)

You mention TypeScript, is this an indication that the issue does not reproduce when you don't set the `--typescript` flag? Can you reproduce the problem without it?

Others have suggested try upgrading, but you seemed to be frustrated that this is only available on `canary`. Today, we released a new `latest`, (`12.0.8` as of writing), so you might try again. If that does not help (considering this issue was reported to already use `12.0.8`), checking `canary` is still a good option, as it would indicate that the problem is indeed already solved, we only need to cut a new `latest` release.

I can also see there is a discussion already opened by you here: #32034

Since this is not a confirmed bug and there is no reliable reproduction, let's continue the discussion there, since it has already started.

I'll reach out to our team to see if anyone has a better idea about what is happening.

Further, I kindly ask you to refrain from negative/demanding phrases and have a read of our [Code of Conduct](https://github.com/vercel/next.js/blob/canary/CODE_OF_CONDUCT.md).
